packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -47,12 +47,26 @@ abstract class ConstrainedLayoutBuilder<ConstraintType extends Constraints> exte
   /// The builder must not return null.
   final Widget Function(BuildContext context, ConstraintType constraints) builder;
 
+  /// Whether [builder] needs to be called again even if the layout constraints
+  /// are the same.
+  ///
+  /// When this widget is rebuilt, it is most likely that we need to call
+  /// [builder] to build this widget's child. However, subclasses may provide
+  /// ways in which the widget can be updated without needing to rebuild the
+  /// child. Such subclasses can use this method to tell the framework when the
+  /// child widget should be rebuilt.
+  @protected
+  bool updateShouldRebuild(covariant ConstrainedLayoutBuilder<ConstraintType> oldWidget) => true;
+
   // updateRenderObject is redundant with the logic in the LayoutBuilderElement below.
 }
 
 class _LayoutBuilderElement<ConstraintType extends Constraints> extends RenderObjectElement {
   _LayoutBuilderElement(ConstrainedLayoutBuilder<ConstraintType> super.widget);
 
+  @override
+  ConstrainedLayoutBuilder<ConstraintType> get widget => super.widget as ConstrainedLayoutBuilder<ConstraintType>;
```

Comment:
While this used to be the style, we now recommend against providing specialized getters for performance reasons because most callers don't care about the specific type (especially on the rebuild hot path) but still have to pay for the cast. Instead, just cast it at the two or three sides where you need the specific type. See also "Specializing the getters" in our docs: https://master-api.flutter.dev/flutter/widgets/RenderObjectElement-class.html.

Reply:
Done